pkg/config/config.go

```diff
@@ -838,10 +836,10 @@ var (
 // CONTAINERS_CONF environment variable.  This is usually done for testing.
 func Default() (*Config, error) {
 	var err error
-	configOnce.Do(func() {
-		config, err = NewConfig("")
```

Comment:
could the reason be that `NewConfig("")` failed but the second caller got no error because error is local to the function?

`err` must be a global as well, so that further calls get its original value instead of `nil`

Reply:
this is how it is used for pkg/cgroups: https://github.com/containers/libpod/blob/master/pkg/cgroups/cgroups_supported.go#L18-L35

Reply:
I made the same conclusion in https://github.com/containers/common/pull/189.

sync.Once is race free but we're using it wrong.

Reply:
Ok I moved the error to global and can not seem to trigger the race,  PTAL